.github/workflows/smoke_test_main.yml

```diff
@@ -0,0 +1,44 @@
+name: Smoke Test Main
+permissions:
+  actions: read
+on:
+  pull_request: {}
+jobs:
+  build_cli:
+    name: Build cli from release
+    runs-on: public-amd64-2xlarge
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.base_ref }}
+
+      - name: Build Cli
+        uses: ./.github/actions/build_cli_linux_target
+        with:
+          target: x86_64-unknown-linux-musl
+          profile: release
+
+      - name: Setup Rust & Cargo
+        uses: ./.github/actions/setup_rust_cargo
+
+      - name: Install Nextest
+        shell: bash
+        run: |
+          cargo install --version 0.9.85 cargo-nextest --locked
+
+      - name: Install cmake
+        shell: bash
+        run: |
+          sudo apt-get install cmake -y
+
+      - name: Run succeeding test
+        env:
+          TRUNK_PUBLIC_API_ADDRESS: https://api.trunk-staging.io
+        shell: bash
+        run: |
+          ./target/x86_64-unknown-linux-musl/release/trunk-analytics-cli test \
```

Comment:
Do we want to run on latest build or the currently latest in the github action. I could see value for both but I think we definitely at least want the latter

Reply:
Since this is triggered on prs, I want it running on the pr branch. Unless I've horribly misunderstood workflows, using the checkout with `ref: ${{ github.base_ref }}` should be putting us there, right?

Reply:
Maybe this needs to be a separate test then, but I would really like to test staging services against what is latest in the GH action

Reply:
Yep, agreed. We want testing of `main` and we want testing of a "pinned" version for both staging and prod.

Additionally, I don't mind a PR smoke test, but I don't think it's as needed right now as the above.

Reply:
Ah, gotcha. I'll pivot this to do that instead and post to channel.